src/components/FormProduct/index.jsx

```diff
@@ -10,25 +15,37 @@ export const FormProduct = () => {
 
   const [product, setProduct] = useState(initialStateProduct);
 
-  const [error, setError] = useState(false);
+  const [error, setError] = useState('');
 
   const handleInputProduct = (e) => {
     setProduct({
       ...product,
       [e.target.name]: e.target.value,
     });
+    setError(e.target.value ? '' : <span aria-live="polite">All fields are required</span>)  //Required field. Immediate validation
   };
-
   const handleSubmitProduct = (e) => {
     e.preventDefault();
-    let { item, nextPurchase, lastPurchasedDate } = product;
+    const { item, nextPurchase, lastPurchasedDate } = product;
 
-    if (item.trim() === '' || nextPurchase === 0) {
-      setError(true);
-      return;
-    }
+    String.prototype.normalizeItem = function () {
```

Comment:
When I opened the browser console then it showed me a **_message_** `"String prototype is read only, properties should not be added  no-extend-native".` It's a **warning**, maybe you can create an external function for your "normalizeItem" method.

![image](https://user-images.githubusercontent.com/8689897/105277186-4f079700-5b79-11eb-8014-a8295e3f2b6b.png)

Reply:
- Hi Ale, it seems like an excellent test case and it would also meet the acceptance criteria. I made the change, thank you very much. 
- I exported the method nomalizaItem to another file. The warning appears the same. It is a question that I have pending to ask. Why does the warning appear if it is a JavaScript functionality?

Reply:
Hi @macosta08 ! Well, I tried with `grapes.` and `Gr.apes` and now I got a validation error. Great! thanks for fix it!

Well, about your function with `String.prototype.normalizeItem` Is it necessary to use this structure? is not possible functions like in `firestore.js` ? I believe the warning is because your only "read" but not modified. On another hand I tried to find some answer about or you can ask to mentors, please!

Reply:
Hi @alxmcr, the recommendations you make are great, they help us a lot to learn and improve, as my colleague explained, it would meet the acceptance criteria. By the way @macosta08, thank you very much for making the clarification and generating the corresponding changes. I see that normalizeItem has already been extracted to the firestore.js file, thank you both very much!

Reply:
Hi @alxmcr, the recommendations you make are great, they help us a lot to learn and improve, as my colleague explained, it would meet the acceptance criteria. You can verify that the warning message is no longer displayed. By the way @macosta08, thank you very much for making the clarification and generating the corresponding changes. I see that normalizeItem has already been extracted to the firestore.js file, thank you both very much!